client/galaxy/scripts/mvc/workflow/workflow-forms.js

```diff
@@ -11,95 +11,85 @@ export class DefaultForm {
-                            Galaxy.emit.debug("tool-form-workflow::postchange()", "Received new model.", data);
-                            process.resolve();
-                        },
-                        error: function(response) {
-                            Galaxy.emit.debug("tool-form-workflow::postchange()", "Refresh request failed.", response);
-                            process.reject();
-                        }
+        const self = this;
+        const node = options.node;
+        this.workflow = options.workflow;
+        this.datatypes = options.datatypes;
+        this._customize(node);
+        this.form = new ToolFormBase({
+            ...node.config_form,
+            text_enable: "Set in Advance",
+            text_disable: "Set at Runtime",
+            narrow: true,
+            initial_errors: true,
+            cls: "ui-portlet-section",
+            postchange(process, form) {
+                const Galaxy = getGalaxyInstance();
+                const options = form.model.attributes;
+                const current_state = {
+                    tool_id: options.id,
+                    tool_version: options.version,
+                    type: "tool",
+                    inputs: Object.assign({}, form.data.create())
+                };
+                Galaxy.emit.debug("tool-form-workflow::postchange()", "Sending current state.", current_state);
+                axios
+                    .post(`${getAppRoot()}api/workflows/build_module`, current_state)
+                    .then(response => {
+                        const data = response.data;
+                        self._customize(data);
+                        self.form.model.set(data.config_form);
+                        self.form.update(data.config_form);
+                        self.form.errors(data.config_form);
+                        // This hasn't modified the workflow, just returned
+                        // module information for the tool to update the workflow
+                        // state stored on the client with. User needs to save
+                        // for this to take effect.
+                        node.update_field_data(data);
+                        Galaxy.emit.debug("tool-form-workflow::postchange()", "Received new model.", data);
+                        process.resolve();
+                    })
+                    .catch(response => {
+                        Galaxy.emit.debug("tool-form-workflow::postchange()", "Refresh request failed.", response);
+                        process.reject();
                     });
-                }
-            })
-        );
+            }
+        });
     }
-    _customize(form) {
-        var options = form.model.attributes;
-        Utils.deepeach(options.inputs, input => {
+    _customize(node) {
+        const inputs = node.config_form.inputs;
+        Utils.deepeach(inputs, input => {
             if (input.type) {
-                input.connectable = true;
```

Comment:
Did you check if that is maybe required for connecting subworkflows ?

Reply:
Is this related to subworkflows? I think this only effects individual input elements in the workflow tool forms.

Reply:
Alright, makes sense!